hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/metrics/MetricsContext.java

```diff
@@ -204,8 +214,40 @@ public void set(long newValue) {
         }
 
         @Override
-        public long get() {
+        protected long get() {
             return VOLATILE_VALUE_UPDATER.get(this);
         }
     }
+
+    private static final class NoopMetric extends AbstractMetric {
```

Comment:
this seems like a micro-optimization to me. We don't disable Jet-level metrics when metrics collection is disabled. I would just remove it.

Reply:
perf impact of metrics when doing an update is pretty much nothing due to use of lazySet. threadSafeMetrics is only useful for async processors. We have many more metrics for example in ProcessorTasklet which is updated more frequently and we don't optimize for those cases.